src/tic_tac_toe/setup.clj

```diff
@@ -19,33 +23,40 @@
   (hash-map :current (human/create-human-player "X")
             :opponent (computer/create-computer-player "O" "X")))
 
-(defn- setup-utils
-  []
-  (let [game-serializer (serializer/create-serializer)
-        game-io (io/create-console-io)]
-    (hash-map :reader (reader/create-reader game-serializer)
-              :writer (writer/create-writer game-serializer)
-              :ui (ui/create-ui game-io)
-              :options options)))
+(defn- setup-location
+  [directory filename utils]
+  (assoc utils :dir directory :file filename))
 
 (defn- setup-saved-game
-  [directory filename utils]
-  ( -> (reader/load-game (:reader utils) directory filename
-                         human/create-human-player
-                         computer/create-computer-player)
-       (merge utils)))
+  [{:keys [reader dir file]
+    {:keys [no-save load-type]} :messages :as utils}]
+  (if (reader/save-exists? reader dir file)
+    (-> (reader/load-game reader dir file
+                           human/create-human-player
+                           computer/create-computer-player)
+        (merge utils))
+    (get-selection load-options no-save load-type utils)))
 
 (defn- setup-new-game
   [utils]
   (merge {:board (setup-board)} (setup-players) utils))
 
-(defn- new-or-saved?
-  [directory filename utils]
-  (if (reader/save-exists? (:reader utils) directory filename)
-    (setup-saved-game directory filename utils)
-    (setup-new-game utils)))
+(def load-options (hash-map :1 setup-new-game :2 setup-saved-game))
+
+(defn setup-utils
+  []
+  (let [game-serializer (serializer/create-serializer)
+        game-io (io/create-console-io)]
+    (hash-map :reader (reader/create-reader game-serializer)
+              :writer (writer/create-writer game-serializer)
+              :ui (ui/create-ui game-io)
+              :messages messages
+              :options game-options)))
 
 (defn setup-game
-  [directory filename]
-  (->> (setup-utils)
-       (new-or-saved? directory filename)))
+  [{:keys [ui]
+    {:keys [welcome load-type]} :messages :as utils}
+    directory filename]
```

Comment:
I think I mentioned this to you the other day, but this complicated destructuring might suggest that we could do some refactoring. Perhaps the data-structure can be flattened or maybe a function can be extracted. Something to think about going forward.

Reply:
^Same for the function with destructuring above.

Reply:
Thank you very much.  I will address it in the next main iteration (multiple saves), as I should be working with a lot of the code where this occurs.